imap_processing/codice/codice_l1a.py

```diff
@@ -309,49 +315,49 @@ def unpack_science_data(self, science_values: str) -> None:
         science_values : str
             A string of binary data representing the science values of the data.
         """
-        self.compression_algorithm = constants.LO_COMPRESSION_ID_LOOKUP[self.view_id]
+        compression_algorithm = constants.LO_COMPRESSION_ID_LOOKUP[self.view_id]
 
         # Decompress the binary string into a list of integers
-        science_values_decompressed = decompress(
-            science_values, self.compression_algorithm
-        )
+        science_values_decompressed = decompress(science_values, compression_algorithm)
 
         # Re-arrange the counter data
         # For CoDICE-lo, data are a 3D arrays with a shape representing
         # [<num_positions>,<num_spin_sectors>,<num_energy_steps>]
         if self.instrument == "lo":
             self.data = np.array(science_values_decompressed, dtype=np.uint32).reshape(
-                self.num_counters,
-                self.num_positions,
-                self.num_spin_sectors,
-                self.num_energy_steps,
+                (
+                    self.num_counters,
+                    self.num_positions,
+                    self.num_spin_sectors,
+                    self.num_energy_steps,
+                )
             )
 
         # For CoDICE-hi, data are a 3D array with a shape representing
         # [<num_energy_steps>,<num_positions>,<num_spin_sectors>]
         elif self.instrument == "hi":
             self.data = np.array(science_values_decompressed, dtype=np.uint32).reshape(
-                self.num_counters,
-                self.num_energy_steps,
-                self.num_positions,
-                self.num_spin_sectors,
+                (
+                    self.num_counters,
+                    self.num_energy_steps,
+                    self.num_positions,
+                    self.num_spin_sectors,
+                )
             )
 
 
 def create_event_dataset(
-    met: list[int], event_data: str, dataset_name: str, data_version: str
+    apid: int, packet: xr.Dataset, data_version: str
 ) -> xr.Dataset:
     """
     Create dataset for event data.
 
     Parameters
     ----------
-    met : list[int]
-        The Mission Elapsed Time of the data.
-    event_data : str
-        A string of binary numbers representing the event data.
-    dataset_name : str
-        The name for the dataset.
+    apid : int
+        The APID of the packet.
+    packet : xarray.Dataset
+        The packet to process.
```

Comment:
full_dataset ?

I think this is the entire packet file you are passing in and not an individual packet.